_LowLevelCode/cpp/test/combine_sqw.tests/combine_sqw.test.cpp

```diff
@@ -10,820 +10,854 @@
-    // }
-    start_buf_pos = 5;
-    reader.get_pix_for_bin(num_bin_in_file - 860, pPix_info, start_buf_pos, pix_start_num, num_bin_pix, false);
-    ASSERT_EQ(pix_start_num, sample_pix_pos[num_bin_in_file - 860]);
-    ASSERT_EQ(num_bin_pix, sample_npix[num_bin_in_file - 860]);
-    // DISABLED
-    // for (size_t i = 0; i<num_bin_pix * 9; i++) {
-    //     ASSERT_EQ(pixels[pix_start_num * 9 + i], pix_buffer[start_buf_pos * 9 + i]);
-    // }
-
-    reader.get_pix_for_bin(num_bin_in_file - 860 + 1, pPix_info, start_buf_pos, pix_start_num, num_bin_pix, false);
-    ASSERT_EQ(pix_start_num, sample_pix_pos[num_bin_in_file - 860 + 1]);
-    ASSERT_EQ(num_bin_pix, sample_npix[num_bin_in_file - 860 + 1]);
-    // DISABLED
-    // for (size_t i = 0; i<num_bin_pix * 9; i++) {
-    //     ASSERT_EQ(pixels[pix_start_num * 9 + i], pix_buffer[start_buf_pos * 9 + i]);
-    // }
-
-    start_buf_pos = 2;
-    reader.get_pix_for_bin(num_bin_in_file - 1, pPix_info, start_buf_pos, pix_start_num, num_bin_pix, false);
-    ASSERT_EQ(pix_start_num, sample_pix_pos[num_bin_in_file - 1]);
-    ASSERT_EQ(num_bin_pix, sample_npix[num_bin_in_file - 1]);
-    for (size_t i = 0; i<num_bin_pix * 9; i++) {
-        ASSERT_EQ(pixels[pix_start_num * 9 + i], pix_buffer[start_buf_pos * 9 + i]);
-    }
-}
+  std::size_t pix_start_num, num_bin_pix, start_buf_pos(0);
+  std::vector<float> pix_buffer(NUM_PIXBLOCK_COLS * 1000);
+  float *pPix_info = &pix_buffer[0];
 
-TEST_F(TestCombineSQW, SQW_Reader_Propagate_Pix_Threads) {
-    sqw_reader reader;
-    fileParameters file_par;
-    file_par.fileName = test_file_name;
-    file_par.file_id = 0;
-    file_par.nbin_start_pos = bin_pos_in_file;
-    file_par.pix_start_pos = pix_pos_in_file;
-    file_par.total_NfileBins = num_bin_in_file;
-    bool initialized(false);
-    try {
-        reader.init(file_par, false, false, 128,1);
-        initialized = true;
-    }
-    catch (...) {}
+  reader.get_pix_for_bin(0, pPix_info, start_buf_pos, pix_start_num,
+                         num_bin_pix, false);
+  EXPECT_EQ(pix_start_num, 0);
+  EXPECT_EQ(num_bin_pix, 3);
+  for (std::size_t i = 0; i < num_bin_pix * NUM_PIXBLOCK_COLS; i++) {
+    EXPECT_EQ(pixels[pix_start_num * NUM_PIXBLOCK_COLS + i], pix_buffer[i]);
+  }
+  // pix buffer have not changed at all
+  reader.get_pix_for_bin(127, pPix_info, start_buf_pos, pix_start_num,
+                         num_bin_pix, false);
+  EXPECT_EQ(pix_start_num, 338);
+  EXPECT_EQ(num_bin_pix, 0);
+
+  reader.get_pix_for_bin(126, pPix_info, start_buf_pos, pix_start_num,
+                         num_bin_pix, false);
+  EXPECT_EQ(pix_start_num, 334);
+  EXPECT_EQ(num_bin_pix, 4);
+  for (std::size_t i = 0; i < num_bin_pix * NUM_PIXBLOCK_COLS; i++) {
+    EXPECT_EQ(pixels[pix_start_num * NUM_PIXBLOCK_COLS + i], pix_buffer[i]);
+  }
 
-    ASSERT_TRUE(initialized);
+  start_buf_pos = 5;
+  reader.get_pix_for_bin(NUM_BINS_IN_FILE - 860, pPix_info, start_buf_pos,
```

Comment:
what is `860`?

Reply:
The first argument is the bin number to get the pixels from, so I guess here we're retrieving the pixel data stored in the 860th bin from the end. I don't know if the 860 is particularly significant.

Perhaps `bin_number = NUM_BINS_IN_FILE - 860;` makes more sense?

Reply:
I'd go with that at least means it's only done once, and the later refs can be rewritten as bin, bin+1.